### PR TITLE
[add] Caches openssf scorecard results into JSON and commits to repo

### LINKED ISSUE #6: [add] Write openssf results into JSON and commit to repo

### PR DESCRIPTION
**Pull request by Riyabelle25**
Fixes #6

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on:
     - cron: "0 0 1 1-12 *"
   push:
     branches:
-      - 'main'
+      - 'test'
```
Comment (ljharb) [inline .github/workflows/main.yml:8]: commenting just so this isn't merged by accident

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -19,3 +19,16 @@ jobs:
       - uses: ./
         with:
           token: ${{ secrets.TOKEN }}
+      - name: switching from HTTPS to SSH
+        run: git remote set-url origin ${{ secrets.SSH }}
```
Comment (ljharb) [inline .github/workflows/main.yml:23]: hm, what's the explanation here? and why is this a secret?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -19,3 +19,16 @@ jobs:
...
+      - name: check for changes
+        run: git status
```
Comment (ljharb) [inline .github/workflows/main.yml:25]:
```suggestion
        run: git status --exit-code
```
i think?

File: .github/workflows/main.yml (hunk 2 above)
Comment (Riyabelle25) [reply .github/workflows/main.yml:23]: It's supposed to be this: `git@github.com:ljharb/repo-report-cache.git`
But, I do not see the point in keeping it a `secret`. Am updating it in the next commit

Comment (ljharb) [reply .github/workflows/main.yml]: does it not work if it's https?

**Pull request merged**